src/audio/eq_fir.c

```diff
@@ -692,33 +675,42 @@ static int eq_fir_prepare(struct comp_dev *dev)
 	if (ret)
 		return ret;
 
-	/* EQ components will only ever have 1 source and 1 sink buffer */
+	/* Get EQ source and buffers. */
 	sourceb = list_first_item(&dev->bsource_list,
 				  struct comp_buffer, sink_list);
 	sinkb = list_first_item(&dev->bsink_list,
 				struct comp_buffer, source_list);
 
-	/* set period bytes and frame format */
-	comp_set_period_bytes(sourceb->source, dev->frames,
-			      &dev->params.frame_fmt, &cd->period_bytes);
+	/* get source data format */
+	comp_set_period_bytes(sourceb->source, dev->frames, &cd->source_format,
+			      &source_period_bytes);
 
-	/* rewrite frame_bytes for all downstream */
-	dev->frame_bytes = cd->period_bytes / dev->frames;
+	/* get sink data format */
+	comp_set_period_bytes(sinkb->sink, dev->frames, &cd->sink_format,
+			      &sink_period_bytes);
+
+	/* Rewrite params format for this component to match the host side. */
+	if (dev->params.direction == SOF_IPC_STREAM_PLAYBACK)
+		dev->params.frame_fmt = cd->source_format;
+	else
+		dev->params.frame_fmt = cd->sink_format;
```

Comment:
@singalsu im wondering if this can be rolled into the comp_set_period_bytes() method. We can probably rename it to something generic like comp_set_copy_params() possibly

Reply:
That's not straightforward. The cd->source_format and sink_format are defined in the component locally and there's no common SOF  header for them.  If I split the format discovery from comp_set_period_bytes() into a separate function the function comp_set_copy_params() could use it to get the source/sink format information. I don't know if that would be nicer than this.

Reply:
@ranj063 I'll leave this undone for next PR but if you like the idea I will try to do for next proposal.